docs/List.md

```diff
@@ -607,31 +607,32 @@ In many cases, you'll need more than simple object manipulation. You'll need to
 
 Here is an example for a Comments exporter, fetching related Posts:
 
-```jsx
+```tsx
 // in CommentList.js
 import { List, downloadCSV } from 'react-admin';
+import type { FetchRelatedRecords } from 'react-admin';
 import jsonExport from 'jsonexport/dist';
 
-const exporter = (records, fetchRelatedRecords) => {
-    // will call dataProvider.getMany('posts', { ids: records.map(record => record.post_id) }), ignoring duplicate and empty post_id
-    fetchRelatedRecords(records, 'post_id', 'posts').then(posts => {
-        const data = records.map(record => ({
-                ...record,
-                post_title: posts[record.post_id].title,
-        }));
-        return jsonExport(data, {
-            headers: ['id', 'post_id', 'post_title', 'body'],
-        }, (err, csv) => {
-            downloadCSV(csv, 'comments');
-        });
+const exporter = async (comments: Comments[], fetchRelatedRecords: FetchRelatedRecords) => {
+    // will call dataProvider.getMany('posts', { ids: records.map(record => record.post_id) }),
+    // ignoring duplicate and empty post_id
+    const posts = await fetchRelatedRecords<Post>(comments, 'post_id', 'posts')
+    const commentsWithPostTitle = comments.map(comment => ({
+            ...comment,
+            post_title: posts[comment.post_id].title,
+    }));
+    return jsonExport(commentsWithPostTitle, {
+        headers: ['id', 'post_id', 'post_title', 'body'],
+    }, (err, csv) => {
+        downloadCSV(csv, 'comments');
     });
 };
 
 const CommentList = () => (
     <List exporter={exporter}>
```

Comment:
We can probably go one step further and make `exporter` typed when passing a type to `List`:
```tsx
<List<Comment> exporter={exporter}>
```

It would be nice